doc/musig-spec.mediawiki

```diff
@@ -76,18 +191,27 @@ The algorithm ''KeySort(pk<sub>1..u</sub>)'' is defined as:
 Input:
 * The number ''u'' of public keys with ''0 < u < 2^32''
 * The public keys ''pk<sub>1..u</sub>'': ''u'' 32-byte arrays
+* The number ''v'' of tweaks with ''0 &le; v < 2^32''
+* The tweaks ''tweak<sub>1..v</sub>'': ''v'' 32-byte arrays
+* The tweak methods ''is_xonly_t<sub>1..v</sub>'' : ''v'' booleans
 
-The algorithm ''KeyAgg(pk<sub>1..u</sub>)'' is defined as:
-* Let ''Q = KeyAggInternal(pk<sub>1..u</sub>)''; fail if that fails.
+The algorithm ''KeyAgg(pk<sub>1..u</sub>, tweak<sub>1..v</sub>, is_xonly_t<sub>1..v</sub>)'' is defined as:
+* Let ''Q,_,_ = KeyAggInternal(pk<sub>1..u</sub>, tweak<sub>1..v</sub>, is_xonly_t<sub>1..v</sub>)''; fail if that fails.
 * Return ''bytes(Q)''.
 
-The algorithm ''KeyAggInternal(pk<sub>1..u</sub>)'' is defined as:
+The algorithm ''KeyAggInternal(pk<sub>1..u</sub>, tweak<sub>1..v</sub>, is_xonly_t<sub>1..v</sub>)'' is defined as:
 * For ''i = 1 .. u'':
 ** Let ''a<sub>i</sub> = KeyAggCoeff(pk<sub>1..u</sub>, pk<sub>i</sub>)''.
 ** Let ''P<sub>i</sub> = point(pk<sub>i</sub>)''; fail if that fails.
-* Let ''Q = a<sub>1</sub>⋅P<sub>1</sub> + a<sub>2</sub>⋅P<sub>1</sub> + ... + a<sub>u</sub>⋅P<sub>u</sub>''
-* Fail if ''is_infinite(Q)''.
-* Return ''Q''.
+* Let ''Q<sub>0</sub> = a<sub>1</sub>⋅P<sub>1</sub> + a<sub>2</sub>⋅P<sub>1</sub> + ... + a<sub>u</sub>⋅P<sub>u</sub>''
+* Fail if ''is_infinite(Q<sub>0</sub>)''.
+* For ''i = 1 .. v'':
+** Let ''Q<sub>i</sub>, g<sub>i-1</sub>, t<sub>i</sub> = Tweak(Q<sub>i-1</sub>, tweak<sub>i</sub>, is_xonly_t<sub>i</sub>)''; fail if that fails
+* If ''v > 0'':
+** Let ''g = g<sub>0</sub>⋅...⋅g<sub>v-1</sub>''
+** Let ''t = sum<sub>i=1</sub><sup>v</sup> t<sub>i</sub>⋅(prod<sub>j=i</sub><sup>v-1</sup> g<sub>j</sub> ) mod n''
```

Comment:
It's not immediately obvious just how nicely this lends itself to implementing as an accumulator for `t`. The sum of product notation is difficult to translate to 'just negate the accumulator conditionally each step'. Perhaps some color text on that topic would be nice. Not necessary, as it can _also_ be implemented correctly by collecting all of the values, and doing the sums at the end.

Reply:
Good point. I think it's worth changing, but would require introducing g and t accumulators (as you suggested initially) and adjusting the explantion. I'll try this out.